expression-tracker/server.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'open3'
+require 'mqtt'
+require 'json'
+require_relative 'helper'
+include Helper
+
+mqtt = false
+mqtt_broker = ENV["MQTT_BROKER"] || "mqtt://localhost"
+mqtt_topic  = ENV["MQTT_TOPIC"]  || "PIT"
+
+# Do we need to connect to MQTT?
+if ARGV[0] == "--mqtt"
+	mqtt = true
+	mqtt_client = MQTT::Client.connect(mqtt_broker)
+end
+
+# Always write to output.csv	
+Thread.new { `./build/bin/FeatureExtraction -device 0 -aus -2Dfp -3Dfp -pdmparams -pose -gaze -of output.csv` }
+sleep 5
+
+
+headers = File.read('openface-headers').split(',').map(&:strip)
+STDOUT.sync = true
+
+# idxssses of headers to output.
+output_headers = select_headers(headers)
+counter = 0
+@last_action = Time.now
+@previous_values = {}
+
+# Simply tail output.csv and either publish on MQTT or print to STDOUT.
+Open3.popen3('tail -f processed/output.csv') do |_stdin, stdout, _stderr, _wait_thr|
+	stdout.each_line do |line|
+		@values = {}
+		line.split(',').map(&:to_f).each_with_index { |n, i| @values[headers[i]] = n }
+
+		reduced      = Hash[output_headers.map { |n| [headers[n], @values[headers[n]]] }]
+		reduced_line = JSON[reduced]
```

Comment:
We could just wrap line 38 in `JSON[]`, but this is where we can switch output format if we need something other than JSON, perhaps XML, YAML, TOML or SuperX++. We could also switch this out for a `format()` function that does some pre-processing of the selected values before sending it out.